crates/common_utils/src/errors.rs

```diff
@@ -8,36 +8,23 @@
 ///
 pub type CustomResult<T, E> = error_stack::Result<T, E>;
 
-macro_rules! impl_error_display {
-    ($st: ident, $arg: tt) => {
-        impl std::fmt::Display for $st {
-            fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-                fmt.write_str(&format!(
-                    "{{ error_type: {:?}, error_description: {} }}",
-                    self, $arg
-                ))
-            }
-        }
-    };
-}
-
-macro_rules! impl_error_type {
-    ($name: ident, $arg: tt) => {
-        #[doc = ""]
-        #[doc = stringify!(Error variant $name)]
-        #[doc = stringify!(Custom error variant for $arg)]
-        #[doc = ""]
-        #[derive(Debug)]
-        pub struct $name;
-
-        impl_error_display!($name, $arg);
-
-        impl std::error::Error for $name {}
-    };
+/// Parsing Errors
+#[derive(Debug, thiserror::Error)]
+pub enum ParsingError {
+    ///Failed to parse enum
+    #[error("Failed to parse enum: {0}")]
+    EnumParseFailure(&'static str),
+    ///Failed to parse struct
+    #[error("Failed to parse struct: {0}")]
+    StructParseFailure(&'static str),
```

Comment:
Do you think we should call these two together as deserialization failure?

So, if a parsing error occurs, it's either during serialization (encoding), deserialization, or an unknown error?

Other than that, looks good to me!

Reply:
The reason for 2 different types is different error messages.

if we club them into a single deserialization variant
say something like

```
#[error("Failed to deserialize {0}")]
DeserializeFailure(&'static str)
```

We lose some info about whether it was enum or struct,
Also I think we should be able to extend the parse errors to parsing beyond serde...

Reply:
Fair enough.